### PR TITLE
plugins: take care of EnvGen not started

### PR DESCRIPTION
**Pull request by sonoro1234**
As #3671 after clang format

### CODE REVIEW

**Comment by brianlheim** [comment]
thanks!

**Comment by sonoro1234** [comment]
I dont understand why travis fail with DO_LINT=true

**Comment by brianlheim** [comment]
did you format your PR after you rebased it? you need to do that as well.

**Comment by brianlheim** [comment]
i don't understand either, the lines look identical to me. maybe a Windows line ending? if you run the formatall command, does anything change?

**Comment by sonoro1234** [comment]
I dont know about this command `formatall`

**Comment by brianlheim** [comment]
https://github.com/supercollider/supercollider/wiki/Cpp-formatting-instructions#possible-workflows-and-scripts

**Comment by sonoro1234** [comment]
I thougth it should be runned from the build directory but
```
C:\supercolliderrepos\scfork\build_engennotstarted>mingw32-make lint
fatal: Not a git repository (or any of the parent directories): .git
This script must be running in a git repo
mingw32-make[3]: *** [tools\CMakeFiles\lint.dir\build.make:56: tools/CMakeFiles/lint] Error 2
mingw32-make[2]: *** [CMakeFiles\Makefile2:1071: tools/CMakeFiles/lint.dir/all] Error 2
mingw32-make[1]: *** [CMakeFiles\Makefile2:1078: tools/CMakeFiles/lint.dir/rule] Error 2
mingw32-make: *** [Makefile:562: lint] Error 2
```

**Comment by sonoro1234** [comment]
Now a a python script from tools
```
C:\supercolliderrepos\scfork\supercollider\tools>clang-format.py lintall
--- server/plugins/LFUGens.cpp
+++ server/plugins/LFUGens.cpp
@@ -2424,7 +2424,7 @@
 static inline bool EnvGen_nextSegment(EnvGen* unit, int& counter, double& level) {
     // if (unit->m_stage == ENVGEN_NOT_STARTED) { return true; } // this fixes doneAction 14, but breaks with
     // EnvGen_next_aa
-    if (unit->m_stage == ENVGEN_NOT_STARTED) {
+    if (unit->m_stage == ENVGEN_NOT_STARTED) {
         counter = INT_MAX;
         return true;
     }
```

What does it mean?

**Comment by sonoro1234** [comment]
Just did formatall to discover that this extra space was not wanted.

**Comment by brianlheim** [comment]
i don't run mingw so i can't help you debug the first problem, but it should work from command prompt. if you want to take a crack at it i'd appreciate it!

**Comment by sonoro1234** [comment]
> i don't run mingw so i can't help you debug the first problem, but it should work from command prompt. if you want to take a crack at it i'd appreciate it!

mingw32-make is not a full make without MSYS so the problem could be that.
Perhaps I will have to dig into python script for knowing...
```
C:\supercolliderrepos\scfork\supercollider\tools>clang-format.py formatall
```
works well and is enough for me.

File: server/plugins/LFUGens.cpp [hunk 1]
```diff
@@ -2424,7 +2424,10 @@ static inline bool check_gate_ar(EnvGen* unit, int i, float& prevGate, float*& g
 static inline bool EnvGen_nextSegment(EnvGen* unit, int& counter, double& level) {
     // if (unit->m_stage == ENVGEN_NOT_STARTED) { return true; } // this fixes doneAction 14, but breaks with
     // EnvGen_next_aa
-
+    if (unit->m_stage == ENVGEN_NOT_STARTED) {
```
Comment (brianlheim) [inline server/plugins/LFUGens.cpp]: have you investigated the note in the comment above? is there any unwanted behavior here with EnvGen_next_aa?

Comment (sonoro1234) [reply server/plugins/LFUGens.cpp]: #3671 is a very long thread, with lot of references, this is relevant for the question:
https://github.com/supercollider/supercollider/issues/3094#issuecomment-380844266

Comment (brianlheim) [reply server/plugins/LFUGens.cpp]: thanks, glad to know you tested it. please delete the outdated comment.

Comment (sonoro1234) [reply server/plugins/LFUGens.cpp]: done

**Comment by sonoro1234** [comment]
Same appveyor fail as in #4410

**Comment by sonoro1234** [comment]
> i don't run mingw so i can't help you debug the first problem, but it should work from command prompt. if you want to take a crack at it i'd appreciate it!

this git complain seems to expect a build directory inside repo but my build directory is out of repo.
`fatal: Not a git repository (or any of the parent directories): .git`

I get the same error running from the command line `git rev-parse --show-toplevel` if out of repo directory.
So this is not mingw related but git related.

**Comment by brianlheim** [comment]
got it. yeah, our instructions say to build in a subdirectory of the fork, so that's what the cmake target expects. btw, it looks like you're creating a new build directory for every PR, which isn't really necessary, you can reuse the same directory.

**Comment by sonoro1234** [comment]
> got it. yeah, our instructions say to build in a subdirectory of the fork, so that's what the cmake target expects. btw, it looks like you're creating a new build directory for every PR, which isn't really necessary, you can reuse the same directory.

Yes, I do that because CMake recommends out of tree builds.
Running the script from command line is enought for me.

**Comment by brianlheim** [comment]
building in a git-ignored directory is more or less the same thing. i wouldn't take cmake's advice for much, lol.

**Pull request merged**